keys/ecc.go

```diff
@@ -26,6 +28,67 @@ var _ ECC = NoECC{}
 func (_ NoECC) AddECC(input []byte) []byte            { return input }
 func (_ NoECC) CheckECC(input []byte) ([]byte, error) { return input, nil }
 
+// CRC16 does the ieee crc16 polynomial check
+type CRC16 struct {
+	Poly  uint16
+	table *crc16.Table
+}
+
+var _ ECC = &CRC16{}
```

Comment:
Minor nit:
```go
var _ ECC = (*CRC16)(nil)
```
so that we don't have randomly allocated global objects lingering.